src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -75,6 +75,14 @@ public ClassName enclosingClassName() {
     return names.subList(1, names.size());
   }
 
+  /** Returns a class enclosed by the same package or class as this class. */
```

Comment:
Maybe say that `this is equivalent to {@code className.enclosingClassName().nestedClassNamed(name)}` so it's not to be confused with `ClassName.get(className.packageName(), name)`